clientv3/client.go

```diff
@@ -169,6 +169,7 @@ func (c *Client) retryConnection(oldConn *grpc.ClientConn, err error) (*grpc.Cli
 		// conn has already been updated
 		return c.conn, nil
 	}
+	oldConn.Close()
```

Comment:
There's an experimental API that returns the status of a connection https://github.com/grpc/grpc-go/blob/master/clientconn.go#L239-L243. Might be useful?